HelgaShiryaeva/assignment1/Anagram.java

```diff
@@ -51,20 +50,20 @@ public boolean areAnagramsByWords(String first, String second) {
         return true;
     }
 
-    private boolean areWordsAnagrams(String first, String second, Map<Character, Integer> characterCountMap) {
-        characterCountMap.clear();
+    private boolean areWordsAnagrams(String first, String second) {
+        Map<Character, Integer> characterCountMap = new HashMap<>();
         for (char c : first.toCharArray()) {
-            Integer count = characterCountMap.getOrDefault(c, 0);
+            int count = characterCountMap.getOrDefault(c, 0);
             characterCountMap.put(c, count + 1);
         }
         for (char c : second.toCharArray()) {
-            Integer count = characterCountMap.get(c);
-            if (count == null || count == 0) {
+            int count = characterCountMap.getOrDefault(c, 0);
+            if (count == 0) {
                 return false;
             }
             characterCountMap.put(c, characterCountMap.get(c) - 1);
```

Comment:
You already have count available in variable. Just `characterCountMap.put(c, count - 1)`.